### ISSUE TITLE
resource/awscc_ec2_ipam_pool requires capitalization on create but API returns lowercase and causes ForceNew

### ISSUE THREAD

**Issue by drewmullen** [hashicorp/terraform-provider-awscc#323]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the Cloudformation Open Coverage Roadmap.

### Affected Resource(s)

* awscc_ec2_ipam_pool

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
variable "operating_regions" {
    default = ["us-east-1", "us-west-2", "us-west-1"]
}

locals {
    operating_regions = toset(concat([data.aws_region.current.name], var.operating_regions))
}
resource "aws_vpc_ipam" "main" {
    description = "AWS IP Address Manager - Main Region: ${data.aws_region.current.name}"
	dynamic operating_regions {
          for_each = local.operating_regions
          content{
		    region_name = operating_regions.value
          }
	}
}

resource "awscc_ec2_ipam_pool" "ipv4_private_default" {
    address_family = "ipv4"
    ipam_scope_id  = aws_vpc_ipam.main.private_default_scope_id
    allocation_default_netmask_length =  28
    allocation_max_netmask_length = null
    allocation_min_netmask_length = null
    auto_import = false
    description = null
    locale = data.aws_region.current.name
    publicly_advertisable = null
    source_ipam_pool_id = null
    allocation_resource_tags = null #[{}]
    provisioned_cidrs = [{
        cidr = "172.2.0.0/16"
    }]
}
```

### Actual Behavior

Error with lowercase:

```
│ Error: Invalid value
│ 
│   with awscc_ec2_ipam_pool.ipv4_private_default,
│   on main.tf line 65, in resource "awscc_ec2_ipam_pool" "ipv4_private_default":
│   65:     address_family = "ipv4"
│ 
│ expected value to be one of [IPv4 IPv6], got ipv4
```

2nd `terraform apply` after adjusting to `IPv4`
```hcl
  # awscc_ec2_ipam_pool.ipv4_private_default must be replaced
-/+ resource "awscc_ec2_ipam_pool" "ipv4_private_default" {
      ~ address_family                    = "ipv4" -> "IPv4" # forces replacement```
```
### Steps to Reproduce

1. `terraform apply` with lowercase `ipv4` <- errors. change to `IPv4` < works
2. `terraform apply` causes a `ForceNew`

### References

* #0000

**Comment by ewbankkit** [comment]
May be fixed when we pull the latest CloudFormation resource type definitions.

**Comment by ewbankkit** [comment]
For **v0.9.0** (2021-12-16 CloudFormation resource type models):

```console
% git diff internal/service/cloudformation/schemas/AWS_EC2_IPAMPool.json
diff --git a/internal/service/cloudformation/schemas/AWS_EC2_IPAMPool.json b/internal/service/cloudformation/schemas/AWS_EC2_IPAMPool.json
index 07788565..c3c6b1e3 100644
--- a/internal/service/cloudformation/schemas/AWS_EC2_IPAMPool.json
+++ b/internal/service/cloudformation/schemas/AWS_EC2_IPAMPool.json
@@ -115,11 +115,7 @@
         },
         "AddressFamily": {
             "description": "The address family of the address space in this pool. Either IPv4 or IPv6.",
-            "type": "string",
-            "enum": [
-                "IPv4",
-                "IPv6"
-            ]
+            "type": "string"
         },
         "AllocationMinNetmaskLength": {
             "description": "The minimum allowed netmask length for allocations made from this pool.",
@@ -162,6 +158,9 @@
         }
     },
     "additionalProperties": false,
+    "propertyTransform": {
+        "/properties/AddressFamily": "$lowercase(AddressFamily)"
+    },
     "required": [
         "IpamScopeId",
         "AddressFamily"
```

Property transforms are not yet supported (#202) but the removal of the enumeration constraints means that you will be able to use the lowercase values `ipv4` and `ipv6`.

**Comment by drewmullen** [comment]
this was resolved with [0.9](https://github.com/hashicorp/terraform-provider-awscc/releases/tag/v0.9.0)! thanks, @ewbankkit !
